core/src/main/java/software/amazon/awssdk/async/AsyncRequestProvider.java

```diff
@@ -62,4 +64,27 @@ static AsyncRequestProvider fromFile(Path path) {
         return FileAsyncRequestProvider.builder().path(path).build();
     }
 
+    /**
+     * Creates an {@link AsyncRequestProvider} that uses a single string as data.
+     *
+     * @param string The string to provide.
+     * @param cs The {@link Charset} to use.
+     * @return Implementation of {@link AsyncRequestProvider} that uses the specified string.
+     * @see SingleByteArrayAsyncRequestProvider
+     */
+    static AsyncRequestProvider fromString(String string, Charset cs) {
+        return new SingleByteArrayAsyncRequestProvider(string.getBytes(cs));
```

Comment:
FileAsyncRequestProvider in this class uses builder pattern. What about doing same for this?

Reply:
See @shorea comments. ;) I

Reply:
My bad, his comments were hidden due to a new revision of PR.

Reply:
All good 😄